connectivity/ble_indication_hs3001/ble_app_indication_HS3001/src/user_custs1_impl.c

```diff
@@ -0,0 +1,218 @@
+ ****************************************************************************************
+ */
+ 
+void user_send_temperature_humidity_ind(void)
+{
+    uint8_t length_temp;
+	  uint8_t length_hum;
+		uint8_t buffer[4];
+		
+    
+#if defined (CFG_USE_INTERNAL_TEMP_SENSOR) && (__DA14531__)
+    
+        adc_config_t temp_config = {
+        .input_mode = ADC_INPUT_MODE_SINGLE_ENDED,
+        .input = ADC_INPUT_SE_TEMP_SENS,
+    };
+        
+    adc_init(&temp_config);
+    
+    int8_t temperature = adc_get_temp();
+    
+    adc_disable();
+#else
+	 
+		Data_Fetch_all (buffer);
+		double temperature = HS3001_get_temperature(buffer);
+		double hum = HS3001_get_humidity(buffer);
+	
+#endif
+    
+	if ((temperature != previous_temperature) || (hum != previous_hum)){
+	
+		previous_temperature = temperature;
+		previous_hum = hum;
+		
+    length_temp = snprintf(temperature_ntf_string, TEMPERATURE_DATA, SNPRINT_FORMAT, temperature);
+		length_hum = snprintf(hum_ntf_string, HUMIDITY_DATA, SNPRINT_FORMAT, hum);
+		
+		//Allocate a new message
+		struct custs1_val_ntf_ind_req* req = KE_MSG_ALLOC_DYN(CUSTS1_VAL_NTF_REQ,
+																prf_get_task_from_id(TASK_ID_CUSTS1),
+																TASK_APP,
+																custs1_val_ntf_ind_req,
+																length_temp);
+																
+		req->conidx = 0;                                        //Connection ID to send the data to (this application can only have one connection(0))
+		req->notification = false;                               //Data is sent as a notification and not as indication
+		req->handle = USER_IDX_TEMPERATURE_VAL_VAL;             //The handle of the characteristic we want to write to
+		req->length = length_temp;                          		    	//Data length in bytes
+		memcpy(req->value, temperature_ntf_string, length_temp);			//Copy the string to the message
+		
+		ke_msg_send(req);                                       //Send the message to the task
+		
+
+
+		req = KE_MSG_ALLOC_DYN(CUSTS1_VAL_NTF_REQ,
+																prf_get_task_from_id(TASK_ID_CUSTS1),
+																TASK_APP,
+																custs1_val_ntf_ind_req,
+																length_hum);
+		
+		req->conidx = 0;                                           //Connection ID to send the data to (this application can only have one connection(0))
+		req->notification = false;                                    //Data is sent as a notification and not as indication
+		req->handle = USER_IDX_HUMIDITY_VAL_VAL;                    //The handle of the characteristic we want to write to
+		req->length = length_hum;                          		    	//Data length in bytes
+		memcpy(req->value, hum_ntf_string, length_hum);			       //Copy the string to the message
+		
+		ke_msg_send(req);                                       //Send the message to the task
+	
+	}
+    
+	timer_temperature_ind = app_easy_timer(INDICATION_DELAY/10, user_send_temperature_humidity_ind); //Set a timer for NOTIFICATION_DELAY ms
```

Comment:
Here, I am using app_easy_timer. This might be because of a little unclear naming here, but what is happening here is exactly what you mentioned. We are fetching humidity and temperature data in certain (arbitrary) interval, then compare it to previous  result and in case of difference we see indication on our server.